engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java

```diff
@@ -543,4 +545,23 @@ private boolean isValidForRestore(final JobRecordValue jobRecord) {
     final var job = jobRecord.getRecord();
     return job.getRecurringTime() > -1 && job.getRetries() > 0;
   }
+
+  /**
+   * Removing jobs that are mistakenly inserted into the backoff column family due to <a
+   * href="https://github.com/camunda/zeebe/issues/15954">this issue</a>
+   */
+  private void removeWrongJobs() {
```

Comment:
:wrench: Maybe we can make the method name more descriptive. Currently, it's not clear what a "wrong" job is. How about `removeJobsWithoutRetries()`?